cmd/gateway/s3/gateway-s3.go

```diff
@@ -97,52 +97,96 @@ EXAMPLES:
 
 // Handler for 'minio gateway s3' command line.
 func s3GatewayMain(ctx *cli.Context) {
+	hosts := ctx.Args()
+	if !ctx.Args().Present() {
+		hosts = []string{"https://s3.amazonaws.com"}
+	}
+
 	// Validate gateway arguments.
-	host := ctx.Args().First()
-	// Validate gateway arguments.
-	logger.FatalIf(minio.ValidateGatewayArguments(ctx.GlobalString("address"), host), "Invalid argument")
+	logger.FatalIf(minio.ValidateGatewayArguments(ctx.GlobalString("address"), hosts...), "Invalid argument")
 
-	minio.StartGateway(ctx, &S3{host})
+	// Start the gateway..
+	minio.StartGateway(ctx, &S3{hosts})
 }
 
 // S3 implements Gateway.
 type S3 struct {
-	host string
+	hosts []string
```

Comment:
@harshavardhana, It is not clear why the user will need to enter several endpoints as arguments when running S3 gateway, can you elaborate more ?

Reply:
This is a precursor change to bring in gateway based namespace expansion.. for now it is not used.